hedgehog/src/Hedgehog/Internal/Report.hs

```diff
@@ -91,9 +91,8 @@ data FailedAnnotation =
 
 data FailureReport =
   FailureReport {
-      failureSize :: !Size
-    , failureSeed :: !Seed
-    , failureShrinks :: !ShrinkCount
+      failureShrinks :: !ShrinkCount
```

Comment:
Is replacing `failureSize` and `failureSeed` with `ShrinkPath` backwards compatible? Happy to try this branch on a few fairly sized hedgehog test-suites to find out, but I thought I should ask anyways. :eyes:

Reply:
Perhaps hspec-hedgehog and/or tasty-hedgehog rely on `FailureReport`(?)

I think if they do, it'll be probably the location*Xyz* fields, so this change isn't going to break anything. /cc @parsonsmatt @dalaing @gwils

Reply:
We're only using `failureLocation` in `hspec-hedgehog` - this shouldn't break us. Thanks for tagging me in 😄

Reply:
Cool!

OTHO, it looks like [`tasty-hedgehog` might be affected](https://github.com/qfpl/tasty-hedgehog/blob/6642c313ca96f355a04c5b9fd99640597189298e/src/Test/Tasty/Hedgehog.hs#L173-L174), though.

Reply:
Looks like this will break us. I'm happy to change to the new type and revise affected older versions. No worries :)
Since there are a couple of libraries downstream though, could we consider perhaps un-"internal"-ing this in a future release? It would be nice (from my perspective at least) if this API were covered under the PVP version number.

Reply:
@gwils, thank you! How would the change look like on your end? I'm not 100% sure about the best way to derive `failureSize` and `failureSeed` from `ShrinkPath`(?) /cc @ChickenProp

Reply:
In F# Hedgehog, I first changed `recheck` to only take a `string`: the `seed` and `size` are encoded as a single `string` (by separating them with an underscore).  Then I added `shrinkPath` to this encoding, which maintains binary compatibility.

Reply:
Ah, so you can't derive them from it. Those precise data points are no longer available. `FailureReport.failureSeed` has been replaced with `Report.reportSeed`, but it's a different seed.

The idea is that instead of calling

```haskell
recheck (failureSize failureReport) (failureSeed failureReport) prop
```

you'd call one of

```haskell
recheckAt (reportSeed report) (SkipToTest $ reportTests report)
recheckAt (reportSeed report) (SkipToShrink (reportTests report) (failureShrinkPath failureReport))
```

The first mostly reproduces the current behaviour, with the advantage that the seed shown to the user can be passed into `HEDGEHOG_SEED` without changing the code at all. (It's not an exact reproduction, because if the given test case passes this time, the new version will check the remaining test cases, while the old version would only have run that single case.)

 The second uses the new behavior, following the shrink path.

---

I see tasty-hedgehog doesn't use `recheck`, but it'll be the same principle. I think the thing to do will be

* Replace the `Size` in `HedgehogReplay` with a `Skip`, noting `skipCompress` and `skipDecompress` for user-friendliness.
* For the `PropertyConfig` at https://github.com/qfpl/tasty-hedgehog/blob/6642c313ca96f355a04c5b9fd99640597189298e/src/Test/Tasty/Hedgehog.hs#L205, add a final argument `maybe Nothing (Just . fst) replay`. (The `Nothing` lets the user set `HEDGEHOG_SKIP` in the environment. `Just SkipNothing` would forbid that.)
* And replace `checkReport config size seed ...` with `checkReport config 0 seed ...`.

Reply:
Thank you, @ChickenProp, and @TysonMN, for the input here. 👍

@gwils, if you can change to the new type and revise affected older versions, I can merge, and do a release on Hackage.

We can add FailureReport to the public API (to be covered by PVP). But in any case, I don't expect another breaking change in this type any time soon.

Reply:
It looks like our bounds are conservative enough already that you should be right to release. Go ahead!
I've created an issue on tasty-hedgehog to support the functionality here https://github.com/qfpl/tasty-hedgehog/issues/61

Reply:
Sounds good, thank you @gwils!